packages/core/base/src/models/paymentElement.ts

```diff
@@ -1,12 +1,14 @@
 import { Currency, Locale, PaymentMethod } from "./types";
 
+// TODO: Remmeber to update this same interface in the Vue component aswell.
+// packages/ui/vue-ui/src/components/CrossmintPaymentElement.vue
 export interface PaymentElement {
     clientId: string;
     mintArgs: Record<string, any>;
     recipient?: string;
-    paymentMethod: PaymentMethod;
-    currency: Currency;
-    locale: Locale;
+    paymentMethod?: PaymentMethod;
+    currency?: Currency;
+    locale?: Locale;
     // TODO: Use checkout event type when it's ready;
-    onEvent: (event: string, metadata: Record<string, any>) => void;
+    onEvent?: (event: string, metadata: Record<string, any>) => void;
```

Comment:
lets make this something like this in the future

`onEvent<K extends keyof CrossmintEventMap>(event: K, callback: CrossmintEventMap[K]): this`

Reply:
Dope! Added